src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs

```diff
@@ -138,5 +139,32 @@ public async Task<IActionResult> ConfirmEmployer(ConfirmEmployerViewModel viewMo
 
             return RedirectToAction("SelectEmployer", new { viewModel.ProviderId });
         }
+
+        [HttpGet]
+        [Route("{cohortReference}/details/delete")]
+        [DasAuthorize(ProviderFeature.ProviderCreateCohortV2)]
+        [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+        public async Task<IActionResult> Delete(DeleteCohortRequest request)
+        {
+            var model = await _modelMapper.Map<DeleteCohortViewModel>(request);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("{cohortReference}/details/delete")]
+        [DasAuthorize(ProviderFeature.ProviderCreateCohortV2)]
+        [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+        public async Task<IActionResult> Delete([FromServices] IAuthenticationService authenticationService, DeleteCohortViewModel viewModel)
+        {
+            if (viewModel.Confirm.Value)
+            {
+                CommitmentsV2.Types.UserInfo userInfo = authenticationService.UserInfo;
```

Comment:
why do we need  to set this explicitly here ?

Reply:
If we don't set this, the history table isn't populated correctly - i.e. the user who made the change.